tests/unit/tuf_repository_service_worker/models/targets/test_crud.py

```diff
@@ -8,35 +8,65 @@
 
-class TestTargetsCrud:
-    def test_create(self):
-        crud.models.RSTUFTargets = pretend.call_recorder(
-            lambda *a, **kw: "fake_db_target"
+class TestCrud:
+    def test_create_roles(self, monkeypatch):
+        monkeypatch.setattr(
+            crud.models,
+            "RSTUFTargetRoles",
+            pretend.call_recorder(lambda *a, **kw: "test_target_roles"),
+        )
+        mocked_db = pretend.stub(
+            add_all=pretend.call_recorder(lambda *a: None),
+            commit=pretend.call_recorder(lambda: None),
+        )
+        test_target = crud.schemas.RSTUFTargetRoleCreate(
+            rolename="bins-0",
+            version=1,
+        )
+        test_result = crud.create_roles(mocked_db, [test_target])
+        assert test_result == ["test_target_roles"]
+        assert mocked_db.add_all.calls == [pretend.call(["test_target_roles"])]
+        assert mocked_db.commit.calls == [pretend.call()]
+
+    def test_create_file(self, monkeypatch):
+        monkeypatch.setattr(
+            crud.models,
+            "RSTUFTargetRoles",
+            pretend.call_recorder(lambda *a, **kw: "test_target_roles"),
+        )
+        monkeypatch.setattr(
+            crud.models,
+            "RSTUFTargetFiles",
+            pretend.call_recorder(lambda *a, **kw: "test_target_files"),
         )
         mocked_db = pretend.stub(
             add=pretend.call_recorder(lambda *a: None),
             commit=pretend.call_recorder(lambda: None),
             refresh=pretend.call_recorder(lambda *a: None),
         )
-        test_target = crud.schemas.TargetsCreate(
+        test_target_file = crud.schemas.RSTUFTargetFileCreate(
             path="file1.tar.gz",
             info={"info": {"k": "v"}},
-            rolename="bins-0",
             published=False,
             action=crud.schemas.TargetAction.ADD,
         )
+        test_target_role = pretend.stub(id=256)
 
-        test_result = crud.create(mocked_db, test_target)
-        assert test_result == "fake_db_target"
-        assert mocked_db.add.calls == [pretend.call("fake_db_target")]
+        test_result = crud.create_file(
+            mocked_db, test_target_file, test_target_role
+        )
+        assert test_result == "test_target_files"
+        assert mocked_db.add.calls == [pretend.call("test_target_files")]
         assert mocked_db.commit.calls == [pretend.call()]
-        assert mocked_db.refresh.calls == [pretend.call("fake_db_target")]
+        assert mocked_db.refresh.calls == [pretend.call("test_target_files")]
 
-    def test_read_unpublished_rolenames(self):
-        crud.models.RSTUFTargets = pretend.stub(
-            published=False, rolename="all-bins"
+    def test_read_roles_with_unpublished_files(self, monkeypatch):
+        monkeypatch.setattr(
+            crud.models, "RSTUFTargetFiles", pretend.stub(published=False)
+        )
+        monkeypatch.setattr(
+            crud.models, "RSTUFTargetRoles", pretend.stub(rolename="bin-0")
```

Comment:
I wonder if I understood.

The `RSTUFTargetRole` object cannot return a list. It represents a row in the table. https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/14fef8bbdb32cf348f1251bd2a2007f6836844fc/repository_service_tuf_worker/models/targets/crud.py#L51

That is possible only in the `.all()` because it is where the ORM query returns all the objects.